### DISCUSSION TITLE
Proposal: Adding In-Place Editing Feature

### DISCUSSION THREAD

**Discussion by razorback16** [LiYanan2004/MarkdownView#25] [Ideas]
Hello everyone,

I've been using the MarkdownView project, and I find it to be an excellent tool for rendering Markdown content in Swift applications. However, I believe that adding an in-place editing feature could greatly enhance its usability and make it an even more versatile tool.

Is it possible to do in-place editing like [Bear 2.0 App](https://blog.bear.app/2023/07/bear-2-is-here/) where the MarkDown code is hidden by default and when a block is edited, the MarkDown code is shown with lighter shade?

Let's discuss this and see if we can collectively work towards making MarkdownView an even more powerful tool for the Swift community.

Looking forward to your thoughts!

**Comment by LiYanan2004** [comment]
You mean when TextField gains focus, we show raw text, and when it loses focus, we render MarkdownView? 

This implementation may have a problem on text continuity where raw text is typically longer than final rendering content due to markdown attributes, which will have bad experience when switching between the two.

To fix that, we can develop a new kinda rendering system which maintains attributes like `*content*`, etc.

**Reply by LiYanan2004** [reply]
https://github.com/LiYanan2004/MarkdownView/assets/37542129/28e61d22-5e5e-4dc0-b0a9-ab4c5fdf5669

**Comment by razorback16** [comment]
We need to think element by element. When we edit one element, the Markdown characters of that element will be visible and otherwise it will only render MarkDown. This is how Bear does it.

https://github.com/LiYanan2004/MarkdownView/assets/5079550/602df365-cc1b-4173-8410-904b4323714c

**Reply by razorback16** [reply]
Some projects have done what you said, (Keeping Markdown Attributes) but that has a bad experience as a reader. It's only good as a raw Markdown Editor

**Reply by LiYanan2004** [reply]
This requires deep understanding in TextViews, both UITextView and NSTextView.
